### PR TITLE
samples: bluetooth: Add build asserts for VS-using samples

### PR DESCRIPTION
**Pull request by jhedberg**
The README files of these samples already indicate that they use the Zephyr-specific HCI vendor extensions, but this may still not be clear enough to some users, as evidenced in #81779. Add build asserts to make sure the samples only build when Zephyr VS extensions have been enabled.

### CODE REVIEW

**Review by kartben — DISMISSED** [review]

File: samples/bluetooth/hci_pwr_ctrl/src/main.c [hunk 1]
```diff
@@ -22,6 +22,9 @@
 #include <zephyr/bluetooth/gatt.h>
 #include <zephyr/bluetooth/services/hrs.h>
 
+BUILD_ASSERT(IS_ENABLED(CONFIG_BT_HAS_HCI_VS),
```
Comment (Thalley) [inline samples/bluetooth/hci_pwr_ctrl/src/main.c:25]: Unrelated to this PR, but there is actually somewhat of a prefix name clash between `BT_HAS_HCI_VS` and `BT_HAS` (Hearing Access Service)

**Review by Thalley — APPROVED** [review]

File: samples/bluetooth/hci_pwr_ctrl/src/main.c (hunk 1 above)
Comment (jhedberg) [reply samples/bluetooth/hci_pwr_ctrl/src/main.c:25]: Good point. I think it'd make sense to swap HAS and HCI, i.e. make it `BT_HCI_HAS_VS_EXT`. But yes, something for a follow-up PR.

Comment (Thalley) [reply samples/bluetooth/hci_pwr_ctrl/src/main.c:25]: https://github.com/zephyrproject-rtos/zephyr/issues/81851

**Comment by jhedberg** [comment]
@kartben could you take another look, and renew your approval if you're still ok? (the only change are the added DT overlays for qemu, since those did not indicate that the configured controller needs to support Zephyr vendor extensions.

**Comment by jhedberg** [comment]
@hermabe @alwa-nordic @sjanc another review/approval is still needed here

**Review by hermabe — APPROVED** [review]

**Comment by hermabe** [comment]
We could potentially add these functions sending the hci commands to the host instead of having the application sending hci commands directly and then hide them behind a host-level kconfig.

**Comment by jhedberg** [comment]
> We could potentially add these functions sending the hci commands to the host instead of having the application sending hci commands directly and then hide them behind a host-level kconfig.

Yes, we could do that, but it would need to be clearly presented as a host API that's strictly only for controllers with the Zephyr VS extensions. Trying to abstract multiple vendor extension command/event sets behind a "unified" host API is likely going to be futile, since there's no standard for features, parameters or semantics. E.g. even in the specific case that prompted this PR (see the link in the description) the "corresponding" HCI command for Silabs is completely different: it sets a global range for all connections, whereas the Zephyr VS command operates on a single connection and is given an exact TX power value.

**Review by kartben — APPROVED** [review]

**Comment by hermabe** [comment]
> > We could potentially add these functions sending the hci commands to the host instead of having the application sending hci commands directly and then hide them behind a host-level kconfig.
> 
> Yes, we could do that, but it would need to be clearly presented as a host API that's strictly only for controllers with the Zephyr VS extensions. Trying to abstract multiple vendor extension command/event sets behind a "unified" host API is likely going to be futile, since there's no standard for features, parameters or semantics. E.g. even in the specific case that prompted this PR (see the link in the description) the "corresponding" HCI command for Silabs is completely different: it sets a global range for all connections, whereas the Zephyr VS command operates on a single connection and is given an exact TX power value.

Yes, it would only be wrappers around the zephyr vs commands. I am not suggesting adding support for out-of-tree vs commands.

**Pull request merged**